src/main/java/Duke/Parser.java

```diff
@@ -0,0 +1,58 @@
+package Duke;
+
+import Duke.Command.*;
+import Duke.Exceptions.InvalidArgumentException;
+import Duke.Tasks.DeleteCommand;
+import Duke.Exceptions.EmptyCommandException;
+import Duke.Exceptions.InvalidCommandException;
+
+import java.sql.SQLOutput;
+import java.time.LocalDate;
+
+public class Parser {
+
+    public static Command parse(String fullCommand) throws EmptyCommandException, InvalidCommandException, InvalidArgumentException {
+        if (fullCommand.equals("")) {
+            throw new EmptyCommandException();
+        }
+
+        switch (fullCommand) {
+            case "bye":
+                return new ExitCommand();
+            case "list":
+                return new ListCommand();
+        }
+
+        String s[] = fullCommand.split(" ", 2);
+        String firstWord = s[0];
+
+        if (firstWord.equals("mark") || firstWord.equals("unmark") || firstWord.equals("delete") || firstWord.equals("todo") ||
+                firstWord.equals("deadline") || firstWord.equals("event")) {
+            if (s.length <= 1) { throw new InvalidArgumentException();} //might wanna chk for firstWord first
+
+            switch (firstWord) {
+                case "mark":
+                    return new MarkCommand(1, Integer.parseInt(s[1]) - 1);
+                case "unmark":
+                    return new MarkCommand(0, Integer.parseInt(s[1]) - 1);
+                case "delete":
+                    return new DeleteCommand(Integer.parseInt(s[1]) - 1);
+                case "todo":
+                    return new AddCommand(TaskType.ToDo, s[1], false);
+                case "deadline":
+                    String st[] = s[1].split(" /by ", 2);
+                    return new AddCommand(TaskType.Deadline, st[0], false, LocalDate.parse(st[1]));
+                case "event":
+                    String stt[] = s[1].split(" /from ", 2);
+                    String sttt[] = stt[1].split(" /to ", 2);
```

Comment:
Don't quite understand `sttt[]` perharps would be better to use names such as `stringToken1`, `stringToken2`?

Reply:
Noted. It was bad naming on my part.